### PR TITLE
Fix signed warnings compiling lib/Sema/ConstraintSystem.h

### PR DESCRIPTION
**Pull request by hughbe**
Int/unsigned int comparison

### CODE REVIEW

File: lib/Sema/ConstraintSystem.h [hunk 1]
```diff
@@ -2482,7 +2482,8 @@ class ConstraintSystem {
         return false;
 
       // Clearly exponential
-      if (TypeCounter < 32 && CountScopes > (1 << TypeCounter))
+      if (TypeCounter < 32 &&
+        CountScopes >(static_cast<unsigned>(1) << TypeCounter))
```
Comment (slavapestov) [inline lib/Sema/ConstraintSystem.h]: Nitpick: space after >

Comment (hughbe) [inline lib/Sema/ConstraintSystem.h]: Fixed

Comment (CodaFi) [inline lib/Sema/ConstraintSystem.h]: One more: align this with the other conjunct.

**Comment by slavapestov** [comment]
Actually isn't 1U a more concise way of writing this?

File: lib/Sema/ConstraintSystem.h (hunk 1 above)
Comment (hughbe) [inline lib/Sema/ConstraintSystem.h]: This is what clang format produced, odd. Anyways, I've used 1U now so this is simpler

**Comment by CodaFi** [comment]
@swift-ci please smoke test and merge.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Pull request merged**